### PR TITLE
Stdenv: Add ENV.libxml2 and ENV.x11 for Linux

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Add `ENV.libxml2` primarily for the use of `test do` blocks. Also add a dummy `ENV.x11` function.

### CODE REVIEW

File: Library/Homebrew/extend/os/linux/extend/ENV/std.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Stdenv
+  # Add libxml2 to CPPFLAGS
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/linux/extend/ENV/std.rb]: Can remove this comment; the code is readable enough for it to be self-explanatory.

File: Library/Homebrew/extend/os/linux/extend/ENV/std.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  def libxml2
+    append "CPPFLAGS", "-I#{Formula["libxml2"].include/"libxml2"}"
+  rescue FormulaUnavailableError
+    nil
+  end
+
+  def x11; end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/linux/extend/ENV/std.rb]: I'd declare this in the base class as there's nothing Linux-specific here.

File: Library/Homebrew/extend/os/extend/ENV/std.rb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 require "extend/ENV/std"
 require "extend/os/mac/extend/ENV/std" if OS.mac?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/extend/ENV/std.rb]: `if`/`elsif` might be nice here if it doesn't contradict what we do elsewhere?

**Comment by sjackman** [comment]
Thanks for the review, Mike! Updated.

**Comment by sjackman** [comment]
Using `Formula["libxml2"]` causes `Homebrew/core` to auto-tap. The auto-tap checks that the syntax of all tapped formulae is valid. That fails because the constant `MacOS` is undefined on Linux.

Here's a few options:
1. Prevent auto-tapping `Homebrew/core` on Linux.
2. Add a stubbed-out implementation of `MacOS` on Linux that returns `false`/`Version::NULL` for all tests.
3. Use `-I#{HOMEBREW_PREFIX}/include/libxml2` rather than `-I#{Formula["libxml2"].include/"libxml2"}`

**Comment by MikeMcQuaid** [comment]
@sjackman Looks like it may be working now.

**Comment by sjackman** [comment]
Well that's mysterious. Does Jenkins keep the tapped Homebrew/core from between runs of CI?

**Comment by MikeMcQuaid** [comment]
> Well that's mysterious. Does Jenkins keep the tapped Homebrew/core between runs of CI?

It does not but I have a filthy hack which makes Linux use the single `testbottest` formula and pretend that it's all of Homebrew/core.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Comment by sjackman** [comment]
Thanks for merging, Mike!

**Pull request merged**